bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/device/feature/CommandHandler.java

```diff
@@ -0,0 +1,2350 @@
+import org.openhab.binding.insteon.internal.utils.HexUtils;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.IncreaseDecreaseType;
+import org.openhab.core.library.types.NextPreviousType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.PlayPauseType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StopMoveType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.types.UpDownType;
+import org.openhab.core.library.unit.ImperialUnits;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A command handler translates an openHAB command into a insteon message
+ *
+ * @author Daniel Pfrommer - Initial contribution
+ * @author Bernd Pfrommer - openHAB 1 insteonplm binding
+ * @author Rob Nielsen - Port to openHAB 2 insteon binding
+ * @author Jeremy Setton - Rewrite insteon binding
+ */
+@NonNullByDefault
+public abstract class CommandHandler extends FeatureBaseHandler {
+    private static final Set<String> SUPPORTED_COMMAND_TYPES = Set.of("DecimalType", "IncreaseDecreaseType",
+            "OnOffType", "NextPreviousType", "PercentType", "PlayPauseType", "QuantityType", "RefreshType",
+            "RewindFastforwardType", "StopMoveType", "StringType", "UpDownType");
+
+    protected final Logger logger = LoggerFactory.getLogger(CommandHandler.class);
+
+    /**
+     * Constructor
+     *
+     * @param f The DeviceFeature for which this command was intended.
+     *            The openHAB commands are issued on an openhab item. The .items files bind
+     *            an openHAB item to a DeviceFeature.
+     */
+    public CommandHandler(DeviceFeature feature) {
+        super(feature);
+    }
+
+    /**
+     * Returns handler id
+     *
+     * @return handler id based on command parameter
+     */
+    public String getId() {
+        return getParameterAsString("command", "default");
+    }
+
+    /**
+     * Returns if handler can handle the openHAB command received
+     *
+     * @param cmd the openhab command received
+     * @return true if can handle
+     */
+    public abstract boolean canHandle(Command cmd);
+
+    /**
+     * Implements what to do when an openHAB command is received
+     *
+     * @param channelUID the channel uid that generated the command
+     * @param config the channel configuration that generated the command
+     * @param cmd the openhab command to handle
+     */
+    public abstract void handleCommand(InsteonChannelConfiguration config, Command cmd);
+
+    //
+    //
+    // ---------------- the various command handlers start here -------------------
+    //
+    //
+
```

Comment:
```suggestion
```

Reply:
Done. I took the opportunity to clean any unnecessary logging.